src/main/java/fi/aalto/cs/apluscourses/model/ExerciseGroup.java

```diff
@@ -19,6 +20,8 @@
   private final String htmlUrl;
   private final boolean isOpen;
   @NotNull
+  private final List<Long> exerciseOrder;
+  @NotNull
   private final List<Exercise> exercises = Collections.synchronizedList(new ArrayList<>());
```

Comment:
`ExerciseGroup` stores exercises in a list order of which is, apparently, arbitrary. Then there is a list of `long`s that specifies the correct order. Could it be possible that when instantiating the class, we would already put the exercises in the correct order, so only one list of exercises would be needed?